cheroot/cli.py

```diff
@@ -77,12 +78,12 @@ def resolve(cls, full_path):
         """Read WSGI app/Gateway path string and import application module."""
         mod_path, _, app_path = full_path.partition(':')
         app = getattr(import_module(mod_path), app_path or 'application')
-
-        with contextlib.suppress(TypeError):
-            if issubclass(app, server.Gateway):
-                return GatewayYo(app)
-
-        return cls(app)
+        # verify that `app` is a class before the issubclass verification,
+        # otherwise a TypeError will be rised.
+        if inspect.isclass(app) and issubclass(app, server.Gateway):
```

Comment:
EAFP is more natural to Python. If you worry about the Py2 compat, I could cherry-pick this https://github.com/cherrypy/cheroot/pull/243/files#diff-544070cbdc8d4e10b3faf72ba309abc6R18-R31 shim to master.

Reply:
Yeah the original approach was a plain `try/except TypeError`  and ignore with a comment on why are we ignoring that exception, but I end up preferring the LBFL in this case. But it seem that the `suppress` context manager is used a lot in the other PR. If that's going to be included, then yeah I could use that here and be done by leaving this fragment as it was.

Reply:
It's in master already.